### PR TITLE
Turn on kubecontainer.Cache in kubelet

### PR DESCRIPTION
**Pull request by yujuhong**

### CODE REVIEW

**Comment by yujuhong** [comment]
This was previously #19436, which got reverted in #19789.
It was reverted due to flaky reboot test failures. I checked the failed builds and found two bugs:
https://github.com/kubernetes/kubernetes/pull/19436#issuecomment-173030536

None of the bugs are directly caused by this PR, but this PR is blocked until the fixes land: #19842, #19838

**Comment by Random-Liu** [comment]
Will apply LGTM when the two fixes land. :)

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -1639,51 +1658,6 @@ func (kl *Kubelet) syncPod(pod *api.Pod, mirrorPod *api.Pod, runningPod kubecont
...
-	if updateType == kubetypes.SyncPodCreate {
-		// This is the first time we are syncing the pod. Record the latency
-		// since kubelet first saw the pod if firstSeenTime is set.
-		if !firstSeenTime.IsZero() {
-			metrics.PodWorkerStartLatency.Observe(metrics.SinceInMicroseconds(firstSeenTime))
-		}
-		// kubelet may have just been restarted. Re-use the last known
-		// apiPodStatus.
-		apiPodStatus = pod.Status
-		apiPodStatus.StartTime = &unversioned.Time{Time: start}
```
Comment (Random-Liu) [inline pkg/kubelet/kubelet.go]: I suddenly found that the `StartTime` is missing here. But we can of course rely on the `SetPodStatus()`.

Comment (yujuhong) [inline pkg/kubelet/kubelet.go]: I think it's unnecessary to set the StartTime here since `SetPodStatus` will take care of that.

Comment (Random-Liu) [inline pkg/kubelet/kubelet.go]: Agree.

**Comment by Random-Liu** [comment]
LGTM, feel free to apply lgtm when the two fixes are merged.

**Comment by alex-mohr** [comment]
Two dependent PRs merged
@k8s-bot unit test this
@k8s-bot test this

**Comment by timothysc** [comment]
Seems like the merge queue is still backed up...

**Comment by timothysc** [comment]
@ncdc could you please label as 1.2-candidate this is the enablement that changes the kubelet performance.

**Comment by timothysc** [comment]
FYI @deads2k hoping to get this in next rebase, looks like dependent items are already in.

**Comment by Random-Liu** [comment]
@k8s-bot test this please.

**Comment by yujuhong** [comment]
@k8s-bot test this

**Pull request merged**